contributors/devel/sig-scheduling/scheduling_code_hierarchy_overview.md

```diff
@@ -0,0 +1,280 @@
+through informers. Or custom plugins may be provided instead of the default ones.
+Known consumers of the scheduling framework:
+- [cluster-autoscaler](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/simulator/scheduler_based_predicates_checker.go#L48-L79)
+- [cluster-capacity](https://github.com/kubernetes-sigs/cluster-capacity/blob/8e9c2dcf3644cb5f73fca3d35d4e22899c265ad5/pkg/framework/simulator.go#L370-L383)
+
+### Assembling the scheduler
+
+The code is located under `pkg/scheduler`.
+This is where implementation of the default scheduler lives.
+Various elements of the scheduler are initialized and put together here:
+- Default scheduling options such as node percentage, initial and maximum backoff, profiles
+- Scheduler cache and queues
+- Scheduling profiles instantiated to tailor a framework for each profile
+  to better suit pod placement (each profile defines a set of plugins to use)
+- Handler functions for getting the next pod for scheduling (`NextPod`) and error handling (`Error`)
+
+The following steps are taken during the process of creating a scheduler instance:
+- Scheduler [cache is initialized](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L206)
+- Both in-tree and out-of-tree registries with plugins are [merged together](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L208-L211)
+- Metrics are [registered](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L232)
+- [Configurator](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L215-L230)
+  building a scheduler instances (wiring the cache, plugin registry,
+  scheduling algorithm and other elements together)
+- Event handlers [are registered](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L273)
+  to allow the scheduler to react on changes in PVs,
+  PVCs, services and other objects relevant for scheduling (eventually,
+  each plugin will define a set of events on which it reacts,
+  see [kubernetes/kubernetes#94009](https://github.com/kubernetes/kubernetes/issues/94009)
+  for more details).
+
+The following diagram shows how individual elements are connected together
+once initialized. Event handlers make sure pods are properly enqueued
+in the [scheduling queues](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-scheduling/scheduler_queues.md),
+the cache is updated with pods and nodes
+as they go (to provide up-to-date snapshot). Scheduling algorithm and the binding cycle
+have the right instances of the framework available (one instance of the framework per a profile).
+
+![Default scheduler architecture](default_scheduler_architecture.png "Default scheduling architecture")
```

Comment:
Is the extra wording "Default scheduling architecture" intentional?

Reply:
Probably not. I wanted to point out a developer can build a custom scheduler which can have a different architecture. The "Default" here refers to kube-scheduler.